dat/CMakeLists.txt

```diff
@@ -0,0 +1,277 @@
+  DEPENDS makedefs ${NLE_DAT}/data.base
+  OUTPUT data
+  COMMAND $<TARGET_FILE:makedefs> -d
+  # HACK see ../util/CMakeLists.txt
+  COMMAND mv ${NLE_DAT}/data ${NLE_DAT_GEN}/
+  WORKING_DIRECTORY ../)
+
+add_custom_command(
+  DEPENDS makedefs ${NLE_DAT}/rumors.tru ${NLE_DAT}/rumors.fal
+  OUTPUT rumors
+  COMMAND $<TARGET_FILE:makedefs> -r
+  COMMAND mv ${NLE_DAT}/rumors ${NLE_DAT_GEN}/
+  WORKING_DIRECTORY ../)
+
+add_custom_command(
+  DEPENDS makedefs ${NLE_DAT}/quest.txt
+  OUTPUT quest.dat
+  COMMAND $<TARGET_FILE:makedefs> -q
+  COMMAND mv ${NLE_DAT}/quest.dat ${NLE_DAT_GEN}/
+  WORKING_DIRECTORY ../)
+
+add_custom_command(
+  DEPENDS makedefs ${NLE_DAT}/oracles.txt
+  OUTPUT oracles
+  COMMAND $<TARGET_FILE:makedefs> -h
+  COMMAND mv ${NLE_DAT}/oracles ${NLE_DAT_GEN}/
+  WORKING_DIRECTORY ../)
+
+add_custom_command(
+  DEPENDS makedefs
+  OUTPUT options
+  COMMAND $<TARGET_FILE:makedefs> -v
+  COMMAND mv ${NLE_DAT}/options ${NLE_DAT_GEN}/
+  COMMAND rm ${NLE_INC}/date.h # We don't need this here
+  WORKING_DIRECTORY ../)
+
+add_custom_command(
+  DEPENDS makedefs dgn_comp ${NLE_DAT}/dungeon.def
+  OUTPUT dungeon.pdf dungeon
+  COMMAND $<TARGET_FILE:makedefs> -e
+  COMMAND $<TARGET_FILE:dgn_comp> ${NLE_DAT}/dungeon.pdf
+  COMMAND mv ${NLE_DAT}/dungeon.pdf ${NLE_DAT}/dungeon ${NLE_DAT_GEN}/
+  WORKING_DIRECTORY ../)
+
+add_custom_command(
+  DEPENDS lev_comp
+  OUTPUT spec_levs
+  COMMAND $<TARGET_FILE:lev_comp> ${SPECIAL_LEVELS}
+  COMMAND ${CMAKE_COMMAND} -E touch spec_levs)
+
+add_custom_command(
+  DEPENDS lev_comp
+  OUTPUT quest_levs
+  COMMAND $<TARGET_FILE:lev_comp> ${QUEST_LEVELS}
+  COMMAND ${CMAKE_COMMAND} -E touch quest_levs)
+
+add_custom_command(
+  OUTPUT perm record logfile xlogfile sysconf
+  COMMAND ${CMAKE_COMMAND} -E touch perm record logfile xlogfile sysconf)
+
+# NOTE: NetHack normally also generates a shell script at ~bin/nethack, however
+# we likely do not need this, as most users currently have straightforward
+# systems settings. It would be awesome though if someone were to run NLE on
+# some old systems with weird pagers and stuff, at some point, eventually...
+# after we solve AI :)
+
+# NOTE: This is missing *.lev, which gets added later in the install decl
+set(ALL_DAT_OUTPUT
+    bogusmon
+    epitaph
+    engrave
+    data
+    rumors
+    quest.dat
+    oracles
+    options
+    dungeon
+    dungeon.pdf
+    spec_levs
+    quest_levs
+    perm
+    record
+    logfile
+    xlogfile
+    sysconf)
+
+set(ALL_DAT_GEN_OUTPUT ${ALL_DAT_OUTPUT})
+list(TRANSFORM ALL_DAT_GEN_OUTPUT PREPEND ${NLE_DAT_GEN}/)
+
+set(ALL_DAT_NOTGEN
+    license
+    symbols
+    help
+    hh
+    cmdhelp
+    history
+    opthelp
+    wizhelp)
+
+# TODO(NN): try to glob .lev automatically
+#
+# NOTE: Using:
+#
+# install(DIRECTORY ${NLE_DAT_GEN} DESTINATION ${INSTDIR} FILES_MATCHING PATTERN
+# "*.lev")
+#
+# results in `$HACKDIR/dat/*.lev` due to CWD. I've tried to get it to work, but
+# given up after a couple of hours :(
```

Comment:
@heiner FYI. :(

Reply:
More context: ideally we want this list to be generated when running make install. This is possible using `install(DIRECTORY ... FILES_MATCHING PATTERN)`, but the resulting install paths are relative to `build/` dir, rather than `build/dat`.